app/services/browser_support.rb

```diff
@@ -0,0 +1,29 @@
+class BrowserSupport
+  @cache = LruRedux::Cache.new(1_000)
+
+  class << self
+    attr_reader :cache
+
+    def supported?(user_agent)
+      return false if user_agent.nil?
+      return true if browser_support_config.nil?
+
+      cache.getset(user_agent) do
+        BrowserslistUseragent::Match.new(
+          browser_support_config,
+          user_agent,
+        ).version?(allow_higher: true)
```

Comment:
Since I was poking around the source, I noticed [the approach for this class's initialization is pretty inefficient](https://github.com/browserslist/browserslist-useragent-ruby/blob/f17624b1c678a34d67038efcb17394b0175317ae/lib/browserslist_useragent/match.rb#L10-L18), since it's restructuring and normalizing the browser support configuration for every user agent, but there doesn't seem to be a publicly-exposed way to parameterize the user agent instead.

I'm rather tempted to do some internal manipulation of the class to get the same effect. Any strong feellings about this @zachmargolis ?

```diff
diff --git a/app/services/browser_support.rb b/app/services/browser_support.rb
index 6691cf286..2ae0f8a3e 100644
--- a/app/services/browser_support.rb
+++ b/app/services/browser_support.rb
@@ -9,14 +9,18 @@ class BrowserSupport
       return true if browser_support_config.nil?
 
-      cache.getset(user_agent) do
-        BrowserslistUseragent::Match.new(
-          browser_support_config,
-          user_agent,
-        ).version?(allow_higher: true)
-      end
+      cache.getset(user_agent) { matcher_supported_version?(user_agent) }
     end
 
     private
 
+    def matcher_supported_version?(user_agent)
+      matcher.instance_variable_set(:@user_agent_string, user_agent)
+      matcher.version?(allow_higher: true)
+    end
+
+    def matcher
+      @matcher ||= BrowserslistUseragent::Match.new(browser_support_config, nil)
+    end
+
     def browser_support_config
       @browser_support_config = begin
```

Reply:
I don't *super* love digging into classes like that, because ideally they could memoize like the `resolver` method that depends on this... If we did maybe we should pin to this exact version in the Gemfile and maybe drop a comment that we are patching it like this

That said, I'd rather put the effort into the [translation layer to our existing browser gem](https://github.com/18F/identity-idp/pull/7909#discussion_r1122203709) in the other comment but that's a lot more involved

Reply:
> That said, I'd rather put the effort into the [translation layer to our existing browser gem](https://github.com/18F/identity-idp/pull/7909#discussion_r1122203709) in the other comment but that's a lot more involved

Okay, Friday afternoon project was to try to implement this, see latest commits.

Reply:
Also, I was motivated toward this implementation because I noticed even more inefficiencies in how the Browserslist gem makes [multiple references](https://github.com/browserslist/browserslist-useragent-ruby/blob/f17624b1c678a34d67038efcb17394b0175317ae/lib/browserslist_useragent/match.rb#L23-L26) to [an unmemoized, non-trivial user agent resolver](https://github.com/browserslist/browserslist-useragent-ruby/blob/f17624b1c678a34d67038efcb17394b0175317ae/lib/browserslist_useragent/match.rb#L37-L39), and its [normalizing map](https://github.com/browserslist/browserslist-useragent-ruby/blob/f17624b1c678a34d67038efcb17394b0175317ae/lib/browserslist_useragent/query_normalizer.rb#L6-L20) is pretty out-of-date (Edge doesn't make an appearance).